CIME/build.py

```diff
@@ -738,44 +738,47 @@ def _build_libraries(
         if not os.path.exists(shared_item):
             os.makedirs(shared_item)
 
-    mpilib = case.get_value("MPILIB")
-    ufs_driver = os.environ.get("UFS_DRIVER")
+    libs = case.get_values("CASE_SUPPORT_LIBRARIES")
+    logger.info(f"libs from case_support_libraries {libs}")
+    build_script = {}
     cpl_in_complist = False
     for l in complist:
         if "cpl" in l:
             cpl_in_complist = True
-    if ufs_driver:
-        logger.info("UFS_DRIVER is set to {}".format(ufs_driver))
-
-    # This is a bit hacky. The host model should define whatever
-    # shared libs it might need.
-    if ufs_driver and ufs_driver == "nems" and not cpl_in_complist:
-        libs = []
-    elif case.get_value("MODEL") == "cesm":
-        libs = ["gptl", "pio", "csm_share"]
-    elif case.get_value("MODEL") == "e3sm":
-        libs = ["gptl", "mct", "spio", "csm_share"]
-    else:
-        libs = ["gptl", "mct", "pio", "csm_share"]
+    if libs is None or len(libs) < 1:
+        mpilib = case.get_value("MPILIB")
+        ufs_driver = os.environ.get("UFS_DRIVER")
+        if ufs_driver:
+            logger.info("UFS_DRIVER is set to {}".format(ufs_driver))
+
+        # This is a bit hacky. The host model should define whatever
+        # shared libs it might need.
+        if ufs_driver and ufs_driver == "nems" and not cpl_in_complist:
+            libs = []
+        elif case.get_value("MODEL") == "cesm":
+            libs = ["gptl", "pio", "csm_share"]
+        elif case.get_value("MODEL") == "e3sm":
+            libs = ["gptl", "mct", "spio", "csm_share"]
+        else:
+            libs = ["gptl", "mct", "pio", "csm_share"]
 
-    libs.append("FTorch")
+        libs.append("FTorch")
 
-    if mpilib == "mpi-serial":
-        libs.insert(0, mpilib)
+        if mpilib == "mpi-serial":
+            libs.insert(0, mpilib)
 
-    if uses_kokkos(case) and comp_interface != "nuopc":
-        libs.append("ekat")
+        if uses_kokkos(case) and comp_interface != "nuopc":
```

Comment:
@bartgol if I understood your suggestion from https://github.com/ESMCI/cime/issues/4837 correctly, the `uses_kokkos` function can be deprecated since the host model will define the shared libs in its config_component.xml?